smartsim/_core/launcher/launcher.py

```diff
@@ -88,6 +88,7 @@ def supported_rs(self) -> t.Dict[t.Type[SettingsBase], t.Type[Step]]:
 
     # every launcher utilizing this interface must have a map
     # of supported RunSettings types (see slurmLauncher.py for ex)
+    @t.final
```

Comment:
This is no longer needed as there is now no special logic when creating a step to prepare for it to be sent through the proxy. Remove this so that subclasses can extend this method (should they choose)